In [9]:
import numpy as np
import pandas as pd

# from the given table observed data is:
observed = np.array([
    [50, 70],   
    [80, 100],  
    [60, 90],   
    [30, 50],   
    [20, 50]    
])

observed

# so converting the observed values into the data frame so that operations can done easily.
df = pd.DataFrame(observed)
df.columns=["smart thermostat","smart light"] # giving the columns names:
df

smart thermostat  smart light
0                50           70
1                80          100
2                60           90
3                30           50
4                20           50

 # 1 hypotheses
  # 1.null hypotheses
 H0 : both the 'smart thermostat' and 'smart light' are independent to each other.

 # 2.alternative hypotheses
 H1 : both the  'smart thermostat' and 'smart light' are depended on each other.

In [26]:
# 2. Compute the Chi-Square Statistic:

import researchpy as rp
from scipy.stats import chi2_contingency
chi2, p_value ,dof,expected = chi2_contingency(observed)
print("Chi-Square Statistic:", chi2)
print("P-value:", p_value)
print("Degrees of Freedom:", dof)

Chi-Square Statistic: 5.638227513227513
P-value: 0.22784371130697179
Degrees of Freedom: 4


C:\Users\dell\anaconda3.1\Lib\site-packages\researchpy\crosstab.py:146: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'Pearson Chi-square ( 12.0) = ' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  table.iloc[0,0] = f"Pearson Chi-square ({dof: .1f}) = "


In [28]:
# 3. Determine the Critical Value:

# to calculate critical value we nead significance value (alpha) and degree of freedom(DF)

alpha = 0.05 # given in the question
q = 0.95
# DF = (n.of rows -1 )*(n.of col -1)
DF = (5-1)*(2-1)
DF

# critical value is:

import scipy.stats as stats
crit = stats .chi2.ppf(q,DF)
crit = (np.round(crit,2))
print("critical value :",crit)

critical value : 9.49


# 4. Make a Decision:

from the above results we got 
Chi-Square Statistic: 5.638227513227513
P-value: 0.22784371130697179
critical value : 9.49

so based on the outcomes we can say that H0 is accepted and H1 is rejected.
that is both the 'smart thermostat' and 'smart light' are independent to each other.
